api/src/main/java/jakarta/config/Config.java

```diff
@@ -118,6 +120,24 @@
      */
     <T> Optional<List<T>> asList(Class<T> type);
 
+    /**
+     * Register a change consumer for this config node. The consumer would only be called if a change in configuration impacts
+     * the current node or its sub-nodes.
+     * <p>
+     * Example 1: If this node is {@code server.host} and {@code server.host} is modified, the consumer would be triggered,
+     * in case {@code server.port} is modified, the consumer would not be triggered.
+     * <p>
+     * Example 2: If this node is {@code server} and either {@code server.host} or {@code server.port} is modified, the
+     * consumer would be triggered.
+     *
+     * @param changeConsumer processor of changes, the config provided to the function is the same node as the current node,
+     *                       but with values from the changed config source, the collection provided is the set of node
+     *                       names that were modified (if this node is {@code server} and {@code server.host} was modified,
+     *                       the set would contain the String {@code host}; the function returns {@code true} if further
+     *                       events should be delivered, {@code false} to stop receiving change events
+     */
+    void onChange(BiFunction<Config, Collection<String>, Boolean> changeConsumer);
```

Comment:
If we want to return a `boolean` here then the better type to use would be `BiPredicate` and avoid boxing on the return.  But, I think it would be better still to introduce a purpose-specific interface like:

```java
@FunctionalInterface
public interface ConfigChangeListener {
    boolean notifyChance(Config config, Collection<String> keys);
}
```

Reply:
It's not clear from the documentation if the collection represents the total set of changed keys, or the sequence of key segments which represent the (single) changed configuration key.